include/mbedtls/block_cipher.h

```diff
@@ -25,7 +25,7 @@
 #endif
 
 #if defined(MBEDTLS_BLOCK_CIPHER_SOME_PSA)
-#include "psa/crypto.h"
+#include "psa/crypto_types.h"
```

Comment:
Hmm, I know I suggested this, but now I'm having doubts: this file says
```
 * \note This file may not be included directly. Applications must
 * include psa/crypto.h. Drivers must include the appropriate driver
 * header file.
```
Is the library allowed to include it directly? @gilles-peskine-arm wdyt?

For context, we're trying to avoid a dependency cycle here:
- `mbedtls_block_cipher_context_t` has fields of types `psa_key_type_t` and `mbedtls_svc_key_id_t`
- if we include `psa/crypto.h`, which is the normal way to get those types defined, then the following will be included: `crypto_struct.h` -> `crypto_driver_contexts_composites.h` -> `gcm.h` + `ccm.h`
- but there the contexts (in configs or interest) have a field of type `mbedtls_block_cipher_context_t` which is not defined because all this happened before we could reach that definition in `block_cipher.h`.

Including `crypto_types.h` here instead of `crypto.h` avoids that issue, but is it the right thing to do?

Reply:
Applications must not include these headers directly because we don't offer any stability guarantees. It's ok for library headers to include them. This specific case seems right to me: the reason `crypto_types.h` has declarations without definitions for struct types is precisely so that it can be included before `crypto_struct.h` is available in the header dependency chain.

Reply:
Great, thanks!